components/modal/w-modal.vue

```diff
@@ -1,30 +1,138 @@
+  const titleCenterClasses = computed(() => ([transitions, titleShouldTransition.value ? 'duration-300' : 'duration-0', props.left ? ccModal.transitionTitleCenter : ccModal.transitionTitleColSpan]));
+  const titleRightClasses = computed(() => ([transitions, titleShouldTransition.value ? 'duration-300' : 'duration-0', ccModal.titleButton, ccModal.titleButtonRight]));
+
+  // when the content area reflows the title area transitions because of v-move in the transition-group
+  // this limits the effects
+  watch(() => ([props.left, props.right, props.title]), async () => {
+    titleShouldTransition.value = true
+    await nextTick()
+    titleShouldTransition.value = false
+  })
+
+  // vue-ism
+  // because we have nested transitions we need to fire this in order
+  // we use nextTick to ensure DOM elements are available for later
+  async function handleTransitions(showing) {
+    if (showing) showModal.value = showing
+    else showContent.value = showing
+    await nextTick()
+    if (showing) showContent.value = showing
+    else showModal.value = showing
+    await nextTick()
+  }
+
+  // change the modal's border radius when within 2% of full height
+  const modifyBorderRadius = () => {
+    if (modalEl.value?.scrollHeight * 1.02 > innerHeight) modalEl.value.style.borderRadius = '0px'
+    else modalEl.value.style.borderRadius = null
+  }
+
+  async function setupHandlers() {
+    focusLock.on(modalEl.value)
+    modalEl.value.addEventListener('transitionend', modifyBorderRadius, { passive: true })
+    addEventListener('keydown', emitIfEscape, { passive: true })
+    setupScrollLock(contentEl.value)
+    emit('shown')
+  }
+  async function teardownHandlers() {
+    // if the modal's parent component is getting unmounted our refs won't exist
+    if (modalEl.value) {
+      focusLock.off(modalEl.value)
+      modalEl.value.removeEventListener('transitionend', modifyBorderRadius, { passive: true })
+    }
+    removeEventListener('keydown', emitIfEscape)
+    teardownScrollLock()
+    emit('hidden')
+  }
+
+  async function handleShow(showing) {
+    if (!windowExists) return
+    if (!showing) await teardownHandlers()
+    await handleTransitions(showing)
+    if (showing) await setupHandlers()
+  }
+
+  watch(() => props.modelValue, handleShow, { immediate: true })
+  onBeforeUnmount(handleShow)
+
+</script>
 <template>
-  <transition name="fade">
-    <div :class="c.backdrop" v-if="showModal" @click.self="emitDismiss" ref="backdropEl" data-test="backdrop">
-      <transition name="slide">
-        <div v-if="showContent" :class="c.modal" tabindex="-1" aria-modal="true" aria-labelledby="w-modal-title" role="dialog" ref="modalEl">
-          <div :class="[c.title, headerClasses]">
-            <transition-group name="w-modal-title">
+  <transition 
+    name="fade"
+    leave-active-class="transition-opacity-200 ease-in-out"
+    enter-active-class="transition-opacity-200 ease-in-out"
```

Comment:
Vue specific transition classes, imho, should not be migrated to the component classes. Wdyt?

Reply:
Why is there vue specific classes in the first place? 🤷 😄

Reply:
🤷 just migrating and applying styles here

Reply:
It would be nice to align everything in the future. Like other stuff 😓

Reply:
Will be odd to have different behaviour on "the same" component on a site, depending on which framework is used... So I think you are correct, leave it out of the component classes... unless it is the desired behaviour? 😄